src/Data/Nat/Divisibility.agda

```diff
@@ -26,43 +24,87 @@ open import Relation.Binary.Definitions
   using (Reflexive; Transitive; Antisymmetric; Decidable)
 import Relation.Binary.Reasoning.Preorder as PreorderReasoning
 open import Relation.Binary.PropositionalEquality.Core
-  using (_≡_; _≢_; refl; sym; trans; cong; cong₂; subst)
+  using (_≡_; refl; sym; cong; cong₂; subst)
 open import Relation.Binary.Reasoning.Syntax
-import Relation.Binary.PropositionalEquality.Properties as PropEq
+open import Relation.Binary.PropositionalEquality.Properties
+  using (isEquivalence; module ≡-Reasoning)
+
+private
+  variable d m n o : ℕ
+
 
 ------------------------------------------------------------------------
--- Definition
+-- Definition and derived properties
 
 open import Data.Nat.Divisibility.Core public
 
+module _ (m∣n : m ∣ n) where
+
+  open _∣_ m∣n
+  open ≤-Reasoning
+
+  instance
+    quotient≢0 : .⦃ NonZero n ⦄ → NonZero quotient
+    quotient≢0 rewrite equality = m*n≢0⇒m≢0 quotient
+
+  quotient∣ : quotient ∣ n
+  quotient∣ = divides m equalityᵒ
+
+  quotient>1 : m < n → 1 < quotient
+  quotient>1 m<n = *-cancelˡ-< m 1 quotient $ begin-strict
+      m * 1        ≡⟨ *-identityʳ m ⟩
+      m            <⟨ m<n ⟩
+      n            ≡⟨ equalityᵒ ⟩
+      m * quotient ∎
+
+  quotient< : 1 < m → .⦃ NonZero n ⦄ → quotient < n
+  quotient< 1<m = begin-strict
+    quotient     <⟨ m<m*n quotient m 1<m ⟩
+    quotient * m ≡⟨ equality ⟨
+    n            ∎
+
+-- defining equation for _/_
+
+  n/m≡quotient : .⦃ _ : NonZero m ⦄ → n / m ≡ quotient
+  n/m≡quotient = *-cancelʳ-≡ _ _ m $ begin-equality
+    n / m * m    ≡⟨ m/n*n≡m m∣n ⟩
+    n            ≡⟨ equality ⟩
+    quotient * m ∎
+
+------------------------------------------------------------------------
+-- Exports
+
+open _∣_ using (quotient) public
```

Comment:
Modulo deciding how much to export as `public` here, I think that this PR is now ready.

Reply:
This export now follows an initial prelude introducing all the new lemmas, and I think, as before, suffices.